core/src/main/scala/org/http4s/otel4s/middleware/ClientMiddleware.scala

```diff
@@ -151,7 +151,7 @@ object ClientMiddleware {
             trace = res.trace
             traceHeaders <- Resource.eval(Tracer[F].propagate(Headers.empty)).mapK(trace)
             newReq = req.withHeaders(traceHeaders ++ req.headers)
-            resp <- poll(client.run(newReq)).guaranteeCase { outcome =>
+            resp <- poll(client.run(newReq)).mapK(trace).guaranteeCase { outcome =>
```

Comment:
A word of warning: I couldn't figure out whether there was any reason to NOT apply `trace` here. Couldn't see any issues during testing, but the explicit use of `trace` below could indicate it was intentional. Leaving this to people with more insights than I have, please review carefully.

Reply:
`trace` is `(fa: F[A]) => Local[F, Context].scope(fa)(spanContext)`, it sets the relevant span context to run the `fa`. That's why applying the trace there is mandatory to allow inner clients to manipulate the context.

Reply:
> but the explicit use of `trace` below could indicate it was intentional

no, this is just a bug that's a result of me not fully understanding the code when I migrated it from natchez. I've actually been meaning to fix it for a while, but I guess you beat me to it lol